### PR TITLE
feat(remasking): connect forward logits to RemaskingPolicy (issue #13)

### PR DESCRIPTION
**Pull request by TomerG711**
## Summary

Implements **[#13 — Connect model forward outputs to RemaskingPolicy inputs](https://github.com/vllm-project/dllm-plugin/issues/13)** for the mock-first MVP: **`vllm_dllm_plugin.remasking.handoff`** validates block logits and `input_draft`, calls **`RemaskingPolicy.apply`** with optional **`remasking_config`**, then **`validate_remask_step_result`**. Intended for **`DllmWorker` (#10)** after last-rank **`compute_logits`**, without **`GPUModelRunner`** forks.

## What’s included

| Area | Details |
|------|---------|
| **API** | `assert_block_logits_shape(logits)` — 2-D `(DRAFT_SIZE, vocab_size)` or length-`DRAFT_SIZE` nested rows; outer length only (per-row vocab width enforced by the policy, documented on the helper). `remask_after_block_forward(..., policy=..., remasking_config=...)` — **`policy` is required**. |
| **Package** | `remasking.__init__` exports `assert_block_logits_shape` and `remask_after_block_forward`. |
| **Docs** | `DESIGN_MVP.md` §5 (forward → remasking); `CONTRACTS.md` (forward handoff, `input_draft` table, §5–8 sync note); `MOCK_STACK_MODEL.md` (block logits vs handoff, `num_tokens == DRAFT_SIZE`); `mock_llada2.py` (call-site note for #10). |
| **Tests** | `tests/test_remask_forward_handoff.py` — guards, parity vs direct `Llada2DefaultRemaskingPolicy.apply`, stub policy proving `policy=` is used, `remasking_config` forwarding (`num_transfer`), tensor cases behind `importorskip("torch")`. |

## Out of scope

- **`DllmWorker` / scheduler** — [#10](https://github.com/vllm-project/dllm-plugin/issues/10) / [#8](https://github.com/vllm-project/dllm-plugin/issues/8).
- **Real weights / Phase 7** — [#12](https://github.com/vllm-project/dllm-plugin/issues/12).

## How to test

```bash
uv sync --group dev
uv run pytest -q
uv sync --group dev --extra vllm
uv run pytest -q tests/test_remask_forward_handoff.py
```

## Commits (high level)

Handoff module, exports, design/contract/mock docs, and handoff tests; refinements from review (required `policy`, stub policy test, `remasking_config` parity).

---

Closes https://github.com/vllm-project/dllm-plugin/issues/13

### CODE REVIEW

**Review by AlonKellner-RedHat — COMMENTED** [review]
**Main critical points:**

1. **Optional default `Llada2DefaultRemaskingPolicy`** — convenient for MVP, but easy for a future non-LLaDA2 stack to misuse without `policy=`; the middle commit that required an explicit policy was safer for a multi-arch plugin. Acceptable if #10 always passes `policy=` from real config.
2. **`test_explicit_policy_passed_through`** still uses the same policy class as the default, so it does not prove the `policy=` argument is honored; a tiny stub policy would be stronger.
3. **`remasking_config`** is forwarded but not covered by these tests (minor if covered elsewhere).
4. **PR description is stale** — it still mentions README handoff/CI notes that the **last commit removes**; update the PR body or squash message before merge.
